src/IPC/QueueRx.cc

```diff
@@ -1,22 +1,74 @@
 #include <iostream>
 #include "IPC.h"
 
+#include "Queue.h"
+
+#include <sys/stat.h>   /* S_IRWXU S_IRWXG */
+#include <sys/ipc.h>    /* IPC_CREAT */
+#include <sys/msg.h>    /* msgget() */
+#include <string.h>     /* strerror() */
+
+char writeBuf[BUFFERSIZE];
+queuedMessage msg;
+
 void QueueRx::receive(void) {
     this->setupQueueRx();
 
-    this->fileSizeRx();
-
     this->queueRx();
 }
 
-void QueueRx::setupQueueRx(void) const {
-    ;
+void QueueRx::setupQueueRx(void) {
+    key_t key;
+
+    /* setup message queue */
+    key = ftok(PathName, ProjectId);    /* key to identify the queue */
+    if (key<0) {
+        throw std::runtime_error(strerror(errno));
+    }
+
+    this->qid = msgget(key, S_IRWXU | S_IRWXG | IPC_CREAT);
+    if (this->qid < 0) {
+        throw std::runtime_error(strerror(errno));
+    }
+
 }
 
-void QueueRx::fileSizeRx(void) const {
-    ;
+void QueueRx::queueRx(void) {
+    long n = 1;
+
+    while (1) {
+
+        if (msgrcv(this->qid, &msg, sizeof(msg), n, MSG_NOERROR | IPC_NOWAIT) < 0) {
+            throw std::runtime_error(strerror(errno));
+        }
+
+        strcpy(writeBuf, msg.payload);
```

Comment:
Everything fine here but to optimize:
Wouldn't it be possible to write msg.payload immediatly to the filestream without passing by a buffer ?